### PR TITLE
Add Coupled Thermo-Mechanical FE Loss with 3D Multi-Output PI-FNO Demonstration

### PR DESCRIPTION
**Pull request by RezaNajian**
## 🚀 Add Coupled ThermoMechanicsLoss + 3D PI-FNO Demo

This PR introduces `ThermoMechanicsLoss`, a monolithic thermo-mechanical finite-element loss coupling conductive heat transfer and linear elasticity with temperature-dependent material response.

The formulation computes element-level residuals and consistent Jacobians for a unified system solving:
- Temperature `T`
- Displacements `(Ux, Uy)` in 2D and `(Ux, Uy, Uz)` in 3D

The implementation is dimension-aware, FE-consistent, and directly compatible with physics-informed training and inverse workflows.

---

## 📓 3D Demonstration: Multi-Output PI-FNO

The notebook `pi_fno_3D_thermomech_demo.ipynb` demonstrates the loss in a fully coupled 3D setting using a **modular multi-output Fourier Neural Operator (FNO)** surrogate.

### ✨ Multi-Output FNO Construction

Instead of using a single multi-channel FNO, the architecture defines:

- **One independent scalar-output FNO per physical field**
- Each FNO predicts exactly **one channel**
- Final multi-field output is obtained via **concatenation of all scalar predictions**

### CODE REVIEW

File: fol/loss_functions/thermo_mechanics.py [hunk 1]
```diff
@@ -0,0 +1,797 @@
...
+        """
+        return self.ComputeElement(xyz[elements_nodes[element_id],:],
+                                         full_control_vector[elements_nodes[element_id]],
+                                         full_dof_vector[((self.number_dofs_per_node*elements_nodes[element_id])[:, jnp.newaxis] +
+                                         jnp.arange(self.number_dofs_per_node))].reshape(-1,1),
+                                         self.thermal_loss_settings["T0"][elements_nodes[element_id]])[0]
+    
+    @print_with_timestamp_and_execution_time
+    @partial(jit, static_argnums=(0,))
+    def ComputeHeatFlux(self,nodal_conductivity: jnp.array, nodal_temperature: jnp.array):
```
Comment (Yusuke-ymzk) [inline fol/loss_functions/thermo_mechanics.py:439]: The L2 projection is fixed according to the definition in addition to minor bug fixes.

File: fol/loss_functions/thermo_mechanics.py [hunk 2]
```diff
@@ -0,0 +1,797 @@
...
+        # Restore original integration rule if modified
+        # -------------------------------------------------
+        if self.loss_settings["num_gp"]<2:
+            self.fe_element.SetGaussIntegrationMethod("GI_GAUSS_1")
+
+        return nodal_flux_vector.reshape(-1,self.dim)
+    
+    @print_with_timestamp_and_execution_time
+    @partial(jit, static_argnums=(0,))    
+    def ComputeStress(self,
```
Comment (Yusuke-ymzk) [inline fol/loss_functions/thermo_mechanics.py:548]: Also here the L2 projection is fixed in addition to minor bug fixes.

File: docs/index.rst [hunk 3]
```diff
@@ -182,4 +182,5 @@ are no longer separate worlds.
 
    installation
    quick_start
-   api_reference/index
\ No newline at end of file
+   api_reference/index
+   3D_thermo_mechanics_example
```
Comment (Yusuke-ymzk) [inline docs/index.rst:186]: A notebook on the 3D thermo-mechanics example is added to the docs.

File: notebooks/pi_fno_3D_thermomech_demo.ipynb [hunk 4]
```diff
@@ -0,0 +1,836 @@
+{
```
Comment (Yusuke-ymzk) [inline notebooks/pi_fno_3D_thermomech_demo.ipynb:1]: The script is modified to be aligned with the script used in the paper. The output is not included in this notebook, as my remote GPU environment is not able to process visualization with PyVista at the moment.

**Review by Yusuke-ymzk — COMMENTED** [review]
The code is updated with some fixes on the features related to the thermo-mechanics example.
